examples/Misc/NavigateSearchResultPagesCachingTokens.php

```
@@ -0,0 +1,270 @@
+    {
+        // The cache of page tokens. It is stored in-memory and in ascendant order of page number.
+        // The first page's token is always an empty string.
+        $pageTokens = [''];
+
+        printf('%s--- 0. Fetch page #1 to get metadata:%1$s%1$s', PHP_EOL);
+
+        // Creates a query that retrieves the campaigns.
+        $query = sprintf(
+            'SELECT campaign.id, campaign.name FROM campaign ORDER BY campaign.name LIMIT %d',
+            self::RESULTS_LIMIT
+        );
+
+        // Issues a paginated search request.
+        $searchOptions = [
+            // Sets the number of results to return per page.
+            'pageSize' => self::PAGE_SIZE,
+            // Requests to return the total results count. This is necessary to determine how many
+            // pages of results there are.
+            'returnTotalResultsCount' => true
+        ];
+        $googleAdsServiceClient = $googleAdsClient->getGoogleAdsServiceClient();
+        $response = $googleAdsServiceClient->search($customerId, $query, $searchOptions);
+        self::cacheNextPageToken($pageTokens, $response->getPage(), 1);
+
+        // Determines the total number of results and prints it.
+        // The total results count does not take into consideration the LIMIT clause of the query
+        // so we need to find the minimal value between the limit and the total results count.
+        $totalNumberOfResults = min(
+            self::RESULTS_LIMIT,
+            $response->getPage()->getResponseObject()->getTotalResultsCount()
+        );
+        printf('Total number of campaigns found: %d.%s', $totalNumberOfResults, PHP_EOL);
+
+        // Determines the total number of pages and prints it.
+        $totalNumberOfPages = ceil($totalNumberOfResults / self::PAGE_SIZE);
+        printf('Total number of pages: %d.%s', $totalNumberOfPages, PHP_EOL);
+        if (!$totalNumberOfPages) {
+            throw new Exception(
```

I think this can be just a normal print and exit the method unless we have a specific reason to throw an error instead.

---

I changed for a normal print because throwing an exception does not bring any value but I use `exit(1)` instead of `return` because I think we should consider the code example to fail when this happens.